test/integration/gridfs/gridfs_stream.test.js

```diff
@@ -874,134 +798,6 @@ describe('GridFS Stream', function () {
     }
   });
 
-  /**
-   * NODE-822 GridFSBucketWriteStream end method does not handle optional parameters
-   */
-  it('should correctly handle calling end function with only a callback', {
-    metadata: { requires: { topology: ['single'] } },
-
-    test(done) {
-      const configuration = this.configuration;
-      const client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
-      client.connect(function (err, client) {
-        const db = client.db(configuration.db);
-        const bucket = new GridFSBucket(db, { bucketName: 'gridfsabort', chunkSizeBytes: 1 });
-        const CHUNKS_COLL = 'gridfsabort.chunks';
-        const uploadStream = bucket.openUploadStream('test.dat');
-
-        const id = uploadStream.id;
-        const query = { files_id: id };
-        uploadStream.write('a', 'utf8', function (error) {
-          expect(error).to.not.exist;
-
-          db.collection(CHUNKS_COLL).count(query, function (error, c) {
-            expect(error).to.not.exist;
-            expect(c).to.equal(1);
-
-            uploadStream.abort(function (error) {
-              expect(error).to.not.exist;
-
-              db.collection(CHUNKS_COLL).count(query, function (error, c) {
-                expect(error).to.not.exist;
-                expect(c).to.equal(0);
-
-                uploadStream.write('b', 'utf8', function (error) {
-                  expect(error.toString()).to.equal('MongoAPIError: Stream has been aborted');
-
-                  uploadStream.end(function (error) {
-                    expect(error.toString()).to.equal('MongoAPIError: Stream has been aborted');
-
-                    // Fail if user tries to abort an aborted stream
-                    uploadStream.abort().then(null, function (error) {
-                      expect(error.toString()).to.equal(
-                        // TODO(NODE-3485): Replace with MongoGridFSStreamClosedError
-                        'MongoAPIError: Cannot call abort() on a stream twice'
-                      );
-                      client.close(done);
-                    });
-                  });
-                });
-              });
-            });
-          });
-        });
-      });
-    }
-  });
-
-  describe('upload stream end()', () => {
-    let client, db;
-
-    afterEach(async () => {
-      sinon.restore();
-      await client.close();
-    });
-
-    it('should not call the callback on repeat calls to end', {
```

Comment:
Calling end more than once makes node throw an error, so we don't have special handling for either of these tests